src/table.jl

```diff
@@ -521,6 +521,11 @@ function uncompress(ptr::Ptr{UInt8}, buffer, compression)
     len = unsafe_load(convert(Ptr{Int64}, ptr))
     ptr += 8 # skip past uncompressed length as Int64
     encodedbytes = unsafe_wrap(Array, ptr, buffer.length - 8)
+    if len === -1
```

Comment:
How do we end up in the `uncompress` method for date that is not compressed?  I believe we should be checking for compression earlier than here.  Is that a branch where this check is missing?

Reply:
I don't know the logic why the writer would do this. I encountered the issue when trying to read files generated from the official java arrow package. When stepping through the debugger during reading in java I encountered this lines which I just copied over to the Julia implementation: https://github.com/apache/arrow/blob/febd0ff144cfb8b2baffb1cb0be57ca40dc7cc77/java/vector/src/main/java/org/apache/arrow/vector/compression/AbstractCompressionCodec.java#L72-L75

In the linked issue there is an example file which also reads in pyarrow. I haven't made any effort to find the corresponding line in there though.

Reply:
Comment about the flag which might shed some light: https://github.com/apache/arrow/blob/cd6e2a4d2b9373b942da18b4cc82cb41431764d9/java/vector/src/main/java/org/apache/arrow/vector/compression/CompressionUtil.java#L67-L71

It could be that the java implementation just does its own thing here without regards to any standard, wouldn't that be a pretty irresponsible thing to do given that the repo markets itself as being cross platform.